solidity/contracts/converter/BancorConverterRegistry.sol

```diff
@@ -76,14 +78,113 @@ contract BancorConverterRegistry is IBancorConverterRegistry, ContractRegistryCl
     }
 
     /**
-      * @dev adds a converter to the registry
-      * anyone can add a converter to the registry, as long as the converter is active and valid
+      * @dev creates a liquid-token converter and adds it to the registry
+      * 
+      * @param _smartTokenName token name
+      * @param _smartTokenSymbol token symbol
+      * @param _smartTokenDecimals token decimals
+      * @param _maxConversionFee maximum conversion-fee
+      * @param _reserveToken reserve token
+      * @param _reserveRatio reserve ratio
+      * @param _reserveAmount reserve amount
+    */
+    function newLiquidToken(
+        string _smartTokenName,
+        string _smartTokenSymbol,
+        uint8 _smartTokenDecimals,
+        uint32 _maxConversionFee,
+        IERC20Token _reserveToken,
+        uint32 _reserveRatio,
+        uint256 _reserveAmount
+    )
+    public payable
+    {
+        IBancorConverterFactory factory = IBancorConverterFactory(addressOf(BANCOR_CONVERTER_FACTORY));
+        SmartToken token = new SmartToken(_smartTokenName, _smartTokenSymbol, _smartTokenDecimals);
+        IBancorConverter converter = IBancorConverter(factory.createConverter(token, registry, _maxConversionFee, IERC20Token(0), 0));

```

Comment:
any way to efficiently check the reserve configuration before creating a new converter? will reduce gas significantly on fail

Reply:
@yudilevi : I'm not sure what the difference is between this comment and the one which says "the contract factory creation is 4M gas - that's what I think we should do after all verifications", but please see updated code.

Reply:
@yudilevi : please see updated code.